lib/salus/scanners/gosec.rb

```diff
@@ -0,0 +1,50 @@
+require 'salus/scanners/base'
+# Gosec scanner check Go for insecure coding patters.
+# https://github.com/securego/gosec
+
+module Salus::Scanners
+  class Gosec < Base
+    def run
+      # Shell Instructions:
+      #   - -fmt=json for JSON output
+      #   - gosec only successfully scans repos within $GOPATH, we force the
+      #     go path reference via symlink within Dockerfile & Dockerfile.tests
+      if ENV['RUNNING_SALUS_TESTS'] == "true"
+        shell_return = run_shell("gosec -fmt=json /go/src/repo/#{@repository.path_to_repo}")
+      else
+        shell_return = run_shell("gosec -fmt=json /go/src/repo")
```

Comment:
Would it be better to recursively scan from this directory if there are multiple folders within a project (via `/go/src/repo/...`)? If so, there might be a problem because symlinks seem to be skipped (https://github.com/securego/gosec/issues/203).

My only workaround for recursive scan was to mount a directory under `/go/src/` and specify it to salus with `-d` option. Hope this helps.

Reply:
will try 👍